lib/impress.application.js

```diff
@@ -332,23 +303,22 @@ impress.application.mixin = (application) => {
     api.fs.readdir(path, (err, files) => {
       if (err) {
         application.log.error(impress.CANT_READ_DIR + relPath);
-        callback(err);
-      } else if (files.length > 0) {
-        api.metasync.each(files, (fileName, cb) => {
-          const fileExt = api.common.fileExt(fileName);
-          const methodName = fileName.substr(0, fileName.length - 3);
-          const filePath = path + '/' + fileName;
-          api.fs.stat(filePath, (err, stats) => {
-            if (!err && stats.isFile() && fileExt === 'js') {
-              application.createScript(filePath, (err, exports) => {
-                apiInterface[methodName] = exports;
-                cb();
-              });
-            } else cb();
+        return callback(err);
+      }
+      if (files.length === 0) return callback(null);
+      api.metasync.each(files, (fileName, cb) => {
+        const fileExt = api.common.fileExt(fileName);
+        const methodName = fileName.substr(0, fileName.length - 3);
+        const filePath = path + '/' + fileName;
+        api.fs.stat(filePath, (err, stats) => {
+          if (err || !stats.isFile() || fileExt !== 'js') return cb();
```

Comment:
1. No error handling.
2. Maybe it's better to perform the `fileExt` check before `fs.stat`?